include/seqan3/alphabet/adaptation/char.hpp

```diff
@@ -0,0 +1,247 @@
+//!\endcond
+struct underlying_rank<char_type>
+{
+    //!\brief An unsigned integer type of the same size as `char_type`.
+    using type = meta::at<detail::char_adaptations_rank_types,
+                          meta::find_index<detail::char_adaptations, char_type>>;
+};
+
+// ------------------------------------------------------------------
+// value metafunctions
+// ------------------------------------------------------------------
+
+//!\brief Specialisation of seqan3::alphabet_size that delegates for char types.
+//!\tparam char_type One of `char`, `char16_t` or `char32_t`.
+//!\sa seqan3::alphabet_size_v
+template <typename char_type>
+//!\cond
+    requires detail::is_char_adaptation_v<char_type>
+//!\endcond
+struct alphabet_size<char_type>
+{
+    //!\brief Smallest unsigned integral type that can hold value;
+    using type = detail::min_viable_uint_t<static_cast<uint64_t>(std::numeric_limits<char_type>::max()) + 1 -
+                                           std::numeric_limits<char_type>::lowest()>;
+    //!\brief The alphabet's size.
+    static constexpr type value =
+        static_cast<type>(std::numeric_limits<char_type>::max()) + 1 - std::numeric_limits<char_type>::lowest();
+};
+
+// ------------------------------------------------------------------
+// free functions
+// ------------------------------------------------------------------
+
+/*!\name Free function wrappers for the char alphabet adaptation
+ * \brief For `char`, `char16_t` and `char32_t` do conversion to/from uint types.
+ * \ingroup adaptation
+ * \{
+ */
+/*!\brief Converting char to char is no-op (it will just return the value you pass in).
+ * \tparam char_type One of `char`, `char16_t` or `char32_t`.
+ * \param chr The alphabet letter that you wish to convert to char.
+ * \returns The letter's value in the alphabet's rank type (usually `char`).
+ */
+template <typename char_type>
+constexpr underlying_char_t<char_type> to_char(char_type const chr)
+    requires detail::is_char_adaptation_v<char_type>
+{
+    return chr;
+}
+
+/*!\brief Convert char to rank by casting to an unsigned integral type of same size.
+ * \tparam char_type One of `char`, `char16_t` or `char32_t`.
+ * \param chr The alphabet letter that you wish to convert to rank.
+ * \returns The letter's value in the alphabet's rank type (usually a `uint*_t`).
+ */
+template <typename char_type>
+constexpr underlying_rank_t<char_type> to_rank(char_type const chr)
+    requires detail::is_char_adaptation_v<char_type>
+{
+    return chr;
+}
+
+/*!\brief Assign a char to the char type (same as calling `=`).
+ * \tparam char_type One of `char`, `char16_t` or `char32_t`.
+ * \param chr The alphabet letter that you wish to assign to.
+ * \param chr2 The `char` value you wish to assign.
+ * \returns A reference to the alphabet letter you passed in.
+ */
+template <typename char_type>
+constexpr char_type & assign_char(char_type & chr, underlying_char_t<char_type> const chr2)
```

Comment:
but it's not a generic alphabet, it's a character. Also the signatures are the only ways to distinguish them in doxygen so it's important to make the parameter and type names suggest something about the overload.